client/context/helpers.go

```diff
@@ -160,7 +176,7 @@ func (ctx CoreContext) SignAndBuild(name, passphrase string, msgs []sdk.Msg, cdc
 }
 
 // sign and build the transaction from the msg
-func (ctx CoreContext) EnsureSignBuildBroadcast(name string, msgs []sdk.Msg, cdc *wire.Codec) (res *ctypes.ResultBroadcastTxCommit, err error) {
+func (ctx CoreContext) EnsureSignBuild(name string, msgs []sdk.Msg, cdc *wire.Codec) (txBytes []byte, err error) {
```

Comment:
Sorry, this is my fault for being insufficiently clear in my last message. I meant could we have ensureSignBuild as a private method which both EnsureSignBuildBroadcast and the async version call. Removing the broadcasting here makes it more work (as you've been changing) in all of the clients. 

Could you bring back EnsureSignBuildBroadcast? Sorry about the confusion / extra work

Reply:
yep i can do that it will be easy